sale_double_validation/models/sale.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, models, _
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.model
+    def _setup_fields(self, partial):
+        super(SaleOrder, self)._setup_fields(partial)
+        selection = self._fields['state'].selection
+        position = 0
+        exists = False
+        for idx, (state, __) in enumerate(selection):
+            if state == 'draft':
+                position = idx
+            elif state == 'to_approve':
+                exists = True
+        if not exists:
+            selection.insert(position + 1, ('to_approve', _('To Approve')))
+
+    @api.multi
+    def is_amount_to_approve(self):
+        currency = self.company_id.currency_id
```

Comment:
`self.ensure_one()`

Reply:
here use `multi ` so not need `self.ensure_one()` only add loop